ci/statistics/databases.py

```diff
@@ -0,0 +1,671 @@
+#     size {integer},
+#     resident {integer},
+#     shared {integer},
+#     text {integer},
+#     data {integer}
+#     );
+#     """
+
+
+# def get_sql_create_mem_table(dbo, name, integer):
+#     create_cpu_hist = f"""
+#     CREATE TABLE {dbo}.{name} (
+#     id {integer},
+#     last                    {integer},
+#     old                     {integer},
+#     older                   {integer},
+#     evenolder                   {integer},
+#     evenmovreolder                   {integer}
+#     );
+#     """
+
+
+# def get_sql_create_mem_table(dbo, name, client):
+#     myquery(client, f"drop table {dbo}.{cpu_table}")
+
+
+# def get_sql_create_mem_table(dbo, name, client):
+#     myquery(client, f"drop table {dbo}.{mem_table}")
+
+
+# def get_sql_create_mem_table(dbo, name, client):
+#     myquery(client, f"drop table {dbo}.{cpu_hist_table}")
+
+
+class MeasurementBase(ABC):
+    """Abstract base class for type Measurements"""
+
+    def __init__(self, lake, name, data_amount, data_length, client, testmode):
+
+        self.data_amount = data_amount
+        self.data_length = data_length
+        self.size = data_length * data_amount
+        self.client = client
+        self.lake = lake
+        self.json_data = None
+        self.job_config = None
+
+        if testmode:
+            self.name = name + "_test"
+        else:
+            self.name = name
+        self.database = f"sturdy-mechanic-312713.ADataSet.{self.name}"
+
+    def postprocess(self, folders, testname, filename, binary):
+        """Postprocess all relevant folders"""
+
+    def show(self):
+        """Show content on console"""
+
+    def update_table(self):
+        """Create table and prepare loading via json and upload"""
+
+    def delete_table(self):
+        """Delete the table from the cloud"""
+        try:
+            self.client.delete_table(self.database)
+        except:  # TODO: Can' import this google.api_core.exceptions.NotFound:
+            pass
+
+    def upload_table(self):
+        """Upload table to online database"""
+
+        if self.client:
+            load_job = self.client.load_table_from_json(
+                self.json_data,
+                self.database,
+                job_config=self.job_config,
+            )
+
+            while load_job.running():
+                time.sleep(0.5)
+                logging.info("Waiting")
+
+            if load_job.errors:
+                logging.error("Error %s", load_job.error_result)
+                logging.error(load_job.errors)
+                raise SystemError
+
+    @staticmethod
+    def foldername_to_index(foldername: str) -> int:
+        """Convert results_N_unpack into N"""
+        reg = re.compile(r"^results_(\d+)_unpack$")
+        match = reg.match(foldername)
+
+        if match:
+            index = int(match.group(1))
+        else:
+            raise SystemError("Cannot convert foldername")
+
+        return index
+
+
+class MeasurementMetadata(MeasurementBase):
```

Comment:
I miss some content description here.

Diving into the code I can have some hints `Tuple[str, str, str, str, str]` or `bigquery.SchemaField("id", "INT64"),`, but it would better to have an understanding of 1) the purpose 2) the schema 3) is this a source dataset or a derived one.